tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,349 @@
+        flowGraph.evaluate();
+
+        assertFalse(flowGraph.getTransposeFlowGraphs()
+                .isEmpty());
+
+        Map<Integer, List<AbstractTransposeFlowGraph>> violatingTransposeFlowGraphs = new HashMap<>();
+        for (var transposeFlowGraph : flowGraph.getTransposeFlowGraphs()) {
+
+            hasLoggingServer(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            hasSecretManager(transposeFlowGraph, violatingTransposeFlowGraphs);
+
+            for (var vertex : transposeFlowGraph.getVertices()) {
+
+                hasGateway(violationsSet, variant, vertex);
+
+                hasAuthenticatedRerquest(violationsSet, variant, vertex);
+
+                hasAuthorizedEntrypoint(violationsSet, variant, vertex);
+
+                hasTransformedEntryIdentity(violationsSet, variant, vertex);
+
+                hasTokenValidation(violationsSet, variant, vertex);
+
+                hasLoginAttemptsRegulation(violationsSet, variant, vertex);
+
+                hasEncryptedEntryConnection(violationsSet, variant, vertex);
+
+                hasEncrytedInternalConnection(violationsSet, variant, vertex);
+
+                hasLocalLogging(violationsSet, variant, vertex);
+
+                hasLogSanitization(violationsSet, variant, vertex);
+
+                hasMessageBroker(violationsSet, variant, vertex);
+
+            }
+        }
+
+        checkCrossTransposeFlowGraphViolations(flowGraph, violatingTransposeFlowGraphs, violationsSet);
+
+    }
+
+    @Test
+    void testConstraints() {
+        for (var model : TUHH_MODELS.keySet()) {
+            for (int variant : TUHH_MODELS.get(model)) {
+                Set<Integer> violationSet = new TreeSet<Integer>();
+                String variationName = model + "_" + variant;
+                performAnalysis(Paths.get(location, model, variationName)
+                        .toString(), variant, violationSet);
+                logger.info("Variant: " + variationName);
+                logger.info("Violations: " + violationSet);
+                assertFalse(violationSet.contains(variant));
+            }
+        }
+    }
+
+    @Test
+    void testDetectCycles() {
+        var model = Paths.get(location, "sqshq", "sqshq_11")
+                .toString();
+
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertTrue(flowGraph.wasCyclic());
+
+    }
+
+    private void checkCrossTransposeFlowGraphViolations(FlowGraphCollection flowGraph,
+            Map<Integer, List<AbstractTransposeFlowGraph>> violatingTransposeFlowGraphs, Set<Integer> violationsSet) {
+        var numOfTransposeFlowGraphs = flowGraph.getTransposeFlowGraphs()
+                .size();
+        // If all TransposeFlowGraphs violate the constraint, the violation is valid
+        // --> if one TransposeFlowGraph has a logging server constraint is not triggered
```

Comment:
Doc: Please extract this comment into the documentation of the complete method (so javadoc)